### PR TITLE
[receiver/k8sobjects] Add ability to exclude deleted updates for watched objects

### LINKED ISSUE #24904: [receiver/k8sobjects] Allow specifying whether DELETE actions should be ingested during a watch
### Component(s)

_No response_

### Is your feature request related to a problem? Please describe.

At the moment the receiver ingests any object update sent to it during a watch.  One of these update types is a `DELETE`.  While occasionally interesting, this type of update is not always needed by a user. For event ingestion, for example, a DELETE option is less useful if the user is only looking for the stream of events that happened in the cluster and is not interested in the cluster's internal management of those events.

### Describe the solution you'd like

A new configuration option that allows excluding `DELETE` types from watches.  We could go 1 step further and allow users to specify exactly which operations (`ADDED`, `UPDATED`, `DELETED`, `BOOKMARK`) the watch should ingest.

### Additional context

A feature like this is necessary for https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24242 since the k8seventsreceiver does not include `DELETE` updates.

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:** 
Adds a new configuration option that allows excluding `DELETED` updates for watched objects.

**Link to tracking Issue:** 
Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24904

**Testing:** 
New unit tests and tested locally

**Documentation:**
Updated the readme

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
I noticed this receiver is using the old style of config testing which makes it difficult to add new scenarios.  I plan to submit another PR to update the testing framework.

**Review by crobert-1 — CHANGES_REQUESTED** [review]
The PR description says the README has been updated, but it's not shown in the changes. Could you add the updated README for review?

File: receiver/k8sobjectsreceiver/README.md [hunk 1]
```diff
@@ -50,6 +50,7 @@ the K8s API server. This can be one of `none` (for no auth), `serviceAccount`
 - `label_selector`: select objects by label(s)
 - `field_selector`: select objects by field(s)
 - `interval`: the interval at which object is pulled, default 60 minutes. Only useful for `pull` mode.
+- `exclude_deleted`: whether the object's deleted updates should be dropped. Only usable in `watch` mode.
```
Comment (dmitryax) [inline receiver/k8sobjectsreceiver/README.md:53]: Why do we need to introduce a config for deleted events only? Shouldn't we have a generic configuration to limit the set of watched events?

Comment (TylerHelmuth) [reply receiver/k8sobjectsreceiver/README.md:53]: You're right, we could make this a generic list of accepted types, which would default to ADDED, UPDATED, and DELETED.

Comment (TylerHelmuth) [reply receiver/k8sobjectsreceiver/README.md:53]: Does k8s allow for custom types returned for a watch? I wouldn't want to miss any that aren't the three I mentioned.

Comment (dmitryax) [reply receiver/k8sobjectsreceiver/README.md:53]: > Does k8s allow for custom types returned for a watch? I wouldn't want to miss any that aren't the three I mentioned.

I'm not sure. @jinja2, do you know by chance?

Comment (TylerHelmuth) [reply receiver/k8sobjectsreceiver/README.md:53]: Maybe the best would be to have it be an exclude list and if it is empty then we exclude nothing. Then we don't have to worry about guessing what is allowed.

Comment (jinja2) [reply receiver/k8sobjectsreceiver/README.md:53]: https://github.com/kubernetes/apimachinery/blob/v0.28.1/pkg/watch/watch.go#L40 should be the possible types

Comment (dmitryax) [reply receiver/k8sobjectsreceiver/README.md:53]: Ok so we can actually use the predefined `EventType`

Comment (TylerHelmuth) [reply receiver/k8sobjectsreceiver/README.md:53]: Updated to allow excluding any watch type.

**Comment by TylerHelmuth** [comment]
running make gotidy cause some changes: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/26314

File: .chloggen/k8sobjects-filter-delete.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: k8sobjectreceiver
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Adds option to excluded `DELETED` updates when in watch mode.
```
Comment (jpkrohling) [inline .chloggen/k8sobjects-filter-delete.yaml:10]:
```suggestion
note: Adds option to exclude `DELETED` updates when in watch mode.
```

File: receiver/k8sobjectsreceiver/receiver_test.go [hunk 3]
```diff
@@ -136,5 +137,71 @@ func TestWatchObject(t *testing.T) {
...
+		rCfg,
+		consumer,
+	)
+
+	ctx := context.Background()
+	require.NoError(t, err)
+	require.NotNil(t, r)
+	require.NoError(t, r.Start(ctx, componenttest.NewNopHost()))
+
+	time.Sleep(time.Millisecond * 100)
```
Comment (jpkrohling) [inline receiver/k8sobjectsreceiver/receiver_test.go:193]: Perhaps an `assert.Never` would be better here?

Comment (TylerHelmuth) [reply receiver/k8sobjectsreceiver/receiver_test.go:193]: The unit tests definitely need some work, but I'd prefer to do it in a separate PR.

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**